lib/Pool.js

```diff
@@ -190,13 +214,16 @@ class Pool {
       // function we returned
       if (typeof callback == 'undefined') {
         let resolveConnectionPromise;
+        let rejectConnectionPromise;
 
         const promise = new Promise((resolve, reject) => {
           resolveConnectionPromise = resolve;
+          rejectConnectionPromise = reject;
         });
         const promiseObj = {
           promise: promise,
-          resolveFunction: resolveConnectionPromise
+          resolveFunction: resolveConnectionPromise,
+          rejectFunction: rejectConnectionPromise
```

Comment:
Do we require ES5+? If so, we could use a trailing comma here and the diff would be cleaner in the future.

Reply:
I think everything in Node is ES5 compliant. I forgot that diffs was the reason for the trailing comma, good call.